### ISSUE TITLE
Feature request: Syntax for alternatives 

### ISSUE THREAD

**Issue by SlimeDog** [SrBedrock/PluginControl#3]
When I am testing alternative resources (for example, DecentHolograms and HolographicDisplays), I have to switch between different versions of the PluginControl configuration, or disable PluginControl temporarily.

#### Describe the solution you'd like

A syntax to permit alternatives. For example (verified legal with [YAML lint](http://www.yamllint.com))
```
enabled: true
plugins:
  - Citizens
  - 
    signs: 
      - CommandSigns
      - ServerSigns
  - ConsoleSpamFix
  - 
    holograms: 
      - DecentHolograms
      - HolographicDisplays
  - DeluxeMenus
```
I prefer legal YAML, but any syntax that permits alternatives would be great.

In this conception, at least one of the alternatives must be enabled. Perhaps a WARN message should be logged if more than one is enabled, but that should be allowed.

#### Describe alternatives you've considered

Switching configurations, as I do now.

**Comment by SlimeDog** [comment]
There are a few inconsistencies with the [Hangar page](https://hangar.papermc.io/SrBedrock/PluginControl) wrt naming.
- The downloaded JAR file is `PluginControl-1.0.2.jar` (without an `s`)
- The contents of `plugin.yml` in the downloaded JAR
  - Commands: `plugincontrol`
  - Permissions: `plugincontrol.use`
- The page URL is `https://hangar.papermc.io/SrBedrock/PluginControl`
- The page title is `PluginsControl` (with a `s`)
- The commands listed on the page are `pluginscontrol *` (with a `s`)
- The permissions listed on the page are `pluginscontrol.use` (with a `s`)

**Comment by SlimeDog** [comment]
I would be happy to test, whenever you're ready.

**Comment by SrBedrock** [comment]
> Any movement on this issue?

Sorry for the delay in getting back to you, I'll be working on this feature today and I'll let you know here when it's ready

**Comment by SlimeDog** [comment]
That's great! Thanks.

**Comment by SrBedrock** [comment]
@SlimeDog can you help me test? 
jar: https://github.com/SrBedrock/PluginControl/suites/16137768582/artifacts/921413552
commands: https://github.com/SrBedrock/PluginControl/issues/3#issuecomment-1596297716

**Comment by SlimeDog** [comment]
Paper 1.20.1-176
PluginControl 1.1.0

Started successfully. After
```
pc group create required-provider
pc group add required-provider PluginControl
pc group add required-provider RequiredPlugins
```
config.yml look like
```
enabled: true
# Action to take if all listed plugins are not enabled.
# log-to-console
# disallow-player-login
# shutdown-server
action: disallow-player-login
kick-message: '[PluginControl] You are not allowed to join the server!'
plugins:
- AcceptTheRules
- BiomeRemap
- Chunky
- ChunkyBorder
- Citizens
- CommandSigns
- ConsoleSpamFix
- DeluxeMenus
- Essentials
- EssentialsChat
- EssentialsGeoIP
- ExtraContexts
- ForceResourcepacks
- ItemJoin
- LuckPerms
- MobColors
- Multiverse-Core
- Multiverse-Inventories
- Multiverse-NetherPortals
- Multiverse-Portals
- MysticalBarriers
- NetworkInterceptor
- pHD
- PlaceholderAPI
- PluginVersions
- ProtocolLib
- Shopkeepers
- Skript
- spark
- TabList
- Vault
- VoidSpawn
- WorldEdit
- WorldGuard
groups:
  required-provider:
  - PluginControl
  - RequiredPlugins
```
and
```
> pc group list
[13:34:37 INFO]: [PluginControl] Groups added: 
Group required-provider [PluginControl and RequiredPlugins]
> pc list
[13:35:00 INFO]: [PluginControl] Plugins added: LuckPerms, WorldEdit, AcceptTheRules, Skript, Chunky, Essentials, MobColors, MysticalBarriers, PlaceholderAPI, PluginVersions, spark, ExtraContexts, ChunkyBorder, CommandSigns, ConsoleSpamFix, NetworkInterceptor, BiomeRemap, Multiverse-Core, ProtocolLib, VoidSpawn, WorldGuard, Citizens, ForceResourcepacks, Shopkeepers, TabList, Multiverse-Inventories, pHD, DeluxeMenus, EssentialsChat, EssentialsGeoIP, ItemJoin, Multiverse-NetherPortals, Multiverse-Portals and Vault
```
.
However, with `PluginVersions` not loaded, I was able to connect to the server as `op`. 
```
> lp group admin permission check plugincontrol.bypass
[14:00:28 INFO]: [LP] Permission information for plugincontrol.bypass:
[14:00:28 INFO]: [LP] - admin does not have plugincontrol.bypass set.
[14:00:28 INFO]: [LP] - admin does not inherit plugincontrol.bypass.
[14:00:28 INFO]: [LP] 
[14:00:28 INFO]: [LP] Permission check for plugincontrol.bypass:
[14:00:28 INFO]: [LP]     Result: undefined
[14:00:28 INFO]: [LP]     Processor: None
[14:00:28 INFO]: [LP]     Cause: None
[14:00:28 INFO]: [LP]     Context: None
```
`bypass`permission was not explicitly or implicitly set. Nevertheless, the connection was allowed. I do not think it should be allowed unless `bypass` permissions is set.

If I demoted myself below `op`, the connection was not allowed, as expected.

It would be nice if the `You are not allowed to join the server!` message was configurable in `lang.yml`.

Extra credit: the results of `pc list` should be in alphabetical order.

**Comment by SlimeDog** [comment]
Feature requests:
- The results of `pc list` and `pc group list` should be displayed in alphabetical order.
- `Plugins added:` should be changed to `Required plugins`.
- `pc list` should also include the group information, after the single list. Otherwise, the list is incomplete.
- `pc list` should identify plugins that are in the list but not enabled -- colored red, for example. Similarly, `pc group list` should identify groups for which no members are loaded.

**Comment by SrBedrock** [comment]
I liked the suggestions, I'll try to add them all as soon as possible

**Comment by SrBedrock** [comment]
> It would be nice if the You are not allowed to join the server! message was configurable in lang.yml

You can change the message in config.yml

**Comment by SlimeDog** [comment]
> You can change the message in config.yml

So I can. I expected it to be in `lang.yml`. Thanks.

**Comment by SrBedrock** [comment]
I believe all the suggestions have been implemented, now kick-message is located in lang.yml and can only be formatted with MiniMessage: https://github.com/SrBedrock/PluginControl/suites/16145602540/artifacts/921855825

**Comment by SlimeDog** [comment]
Everything not identified separately has been implemented in the second PC 1.1.0. Thanks.
